go/tools/builders/generate_test_main.go

```diff
@@ -182,11 +183,14 @@ func main() {
 	}
 
 	{{if .Coverage}}
-	if len(coverdata.Cover.Counters) > 0 {
-		testing.RegisterCover(coverdata.Cover)
-	}
-	if coverageDat, ok := os.LookupEnv("COVERAGE_OUTPUT_FILE"); ok {
-		if testing.CoverMode() != "" {
+	if len(coverdata.Counters) > 0 {
+		testing.RegisterCover(testing.Cover{
+			Mode: "{{ .CoverMode }}",
```

Comment:
Can you use `testing.CoverMode()` here instead of passing it through command line arguments?

Reply:
No, it's not possible. `testing.CoverMode()` returns the `Cover.Mode` defined by the object passed to `testing.RegisterCover()`. Without a call of `testing.RegisterCover()`, `testing.CoverMode()` will always return `""`.